assets/js/datepicker.js

```diff
@@ -1,26 +1,27 @@
 /* global job_manager_datepicker */
-jQuery(document).ready(function($) {
+jQuery(document).on( 'ready wpJobManagerFieldAdded', function() {
```

Comment:
Using  the same callback for these means all the date fields will be initialized again when the new event is triggered. Is that safe, could it lead to problems? Should we extract the callback that initializes a single input, and run only that for new fields?

(Meaning something like: )
`jQuery( 'input.job-manager-datepicker, input#_job_expires' ).each( initializeDatepicker )`)
`jQuery(document).on( 'wpJobManagerFieldAdded' , initializeDatepicker )`

Reply:
Great point @yscik! Missed that! So I am not sure if this actually caused an issue (although I am pretty sure that it did) as I found out another issue where if you added dynamically more than one dates and tried to select the date it would break.

The problem is that datepicker relies on the HTML id and unfortunately the ids that were generated for this form were not unique. I have updated the code to have unique ids, can you have another look please?

I tried another approach initially, to rename the actual fields but it wasn't possible as these are stored in the db and we would break backward compatibility. There are updates in this PR too: https://github.com/Automattic/wpjm-addons/pull/115